lib/marathon_client.ex

```diff
@@ -24,33 +24,55 @@ defmodule MarathonEventExporter.MarathonClient do
     @doc """
     Starts a new StreamCatcher.
     """
-    def start_link({url, listeners}, opts \\ []) do
-      GenServer.start_link(__MODULE__, {url, listeners}, opts)
+    def start_link({url, listeners, timeout}, opts \\ []) do
+      GenServer.start_link(__MODULE__, {url, listeners, timeout}, opts)
     end
 
-
     ## Server callbacks
 
-    def init({url, listeners}) do
+    def init({url, listeners, timeout}) do
```

Comment:
I'm super cautious about timeouts after my Twisted experiences. So I'd like to be clear about what exactly this timeout means. But maybe in a later PR we can work on timeouts specifically and leave this for now.

Reply:
I haven't looked at the implementation under the hood, but the exhibited behaviour is that that we receive an `%HTTPoison.Error{}` struct with `{:closed, :timeout}` as the reason. We handle that by returning `{:stop, reason, state}` from our `handle_info` function, which shuts down the `GenServer` and (because the reason isn't `:normal`) any linked processes. I would expect hackney to do the necessary cleanup at that point, and unlike Erlang/Elixir makes that a whole lot easier than Python+Twisted does.

I should also note that there are two timeout parameters we can specify in hackney calls. `:timeout` (which we leave at the default) is the connect timeout and `:recv_timeout` (which we set here) is the "time between response bytes arriving" that we care about for the stream.

We could also have set the hackney timeout high and used timers in the `GenServer` to have more fine-grained control, but that would be a bunch more work and isn't really necessary unless we run into some kind of problem with the hackney timeouts.

Reply:
Cool, I didn't really manage to find good timeout docs for HTTPoison. So I wasn't sure what exactly `recv_timeout` was.

Reply:
Yeah. A lot of those parameters are just passed directly through to hackney, so you need to read the hackney docs to understand them. Apart from the high-level API bits, I've mostly been working from the hackney docs for this stuff.